### PR TITLE
Remove: T3 env

### PR DESCRIPTION
**Pull request by bhavaygoel**
Removed T3 env and domain check during auth.
Added .env.example

### CODE REVIEW

File: tsconfig.tsbuildinfo
Comment (Maan2003) [inline tsconfig.tsbuildinfo:1]: what is this

File: tsconfig.json [hunk 1]
```diff
@@ -16,15 +16,15 @@
...
       "@/assets/*": ["./public/assets/*"],
-      "@/*": ["./*"]
-    }
+      "@/*": ["./*"],
+    },
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "env.mjs"],
-  "exclude": ["node_modules"]
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "exclude": ["node_modules"],
```
Comment (Maan2003) [inline tsconfig.json:29]: trailing commas are allowed in json?

File: next.config.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import "./env/server.ts"
-
+//@ts-check
```
Comment (Maan2003) [inline next.config.js:1]: maybe, unsure if this works though
```suggestion
// @ts-check
```

File: app/api/auth/auth.ts [hunk 3]
```diff
@@ -1,19 +1,10 @@
...
-      const isAllowedToSignIn = profile.hd == 'nitkkr.ac.in';
-      return isAllowedToSignIn;
-    },
-  },
   providers: [
     GoogleProvider({
-      clientId: env.GOOGLE_CLIENT_ID!,
-      clientSecret: env.GOOGLE_CLIENT_SECRET!,
+      clientId: process.env.GOOGLE_CLIENT_ID!,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
```
Comment (Maan2003) [inline app/api/auth/auth.ts:7]: what is error message

File: .env.example [hunk 4]
```diff
@@ -0,0 +1,4 @@
+GOOGLE_CLIENT_ID =
+GOOGLE_CLIENT_SECRET =
+NEXTAUTH_SECRET =
+NEXTAUTH_URL = "http://localhost:3000/"
```
Comment (Maan2003) [inline .env.example:4]: .env usually doesn't have spaces around `=`, so that it is sourcable in bash

**Review by Maan2003 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 .next/
-.env
\ No newline at end of file
+.env
+tsconfig.tsbuildinfo
```
Comment (Maan2003) [inline .gitignore:4]: new lines :/

File: tsconfig.json (hunk 1 above)
Comment (bhavaygoel) [reply tsconfig.json:29]: prettier added these, i was also confused

File: next.config.js (hunk 2 above)
Comment (GetPsyched) [reply next.config.js:1]: I guess that line is not needed at all for now

File: .gitignore (hunk 5 above)
Comment (GetPsyched) [reply .gitignore:4]: I plan to refactor this later with segregations between what's ignored for what.

File: tsconfig.json (hunk 1 above)
Comment (Maan2003) [reply tsconfig.json:29]: yea ts has custom json format

File: app/api/auth/auth.ts (hunk 3 above)
Comment (GetPsyched) [reply app/api/auth/auth.ts:7]: The env import was removed, so there is no `env` namespace anymore. It's the default `process.env` in JS

Comment (Maan2003) [reply app/api/auth/auth.ts:7]: if process.env.GOOGLE_CLIENT_ID doesn't exist, we should fail with a good error message

Comment (GetPsyched) [reply app/api/auth/auth.ts:7]: Yeah, the `!` is wrong, we should error out properly

Comment (bhavaygoel) [reply app/api/auth/auth.ts:7]: could add a check before accessing and if not found, throw error

**Review by GetPsyched — APPROVED** [review]
LGTM

**Pull request merged**